tests/functional/conftest.py

```diff
@@ -63,17 +63,9 @@ def with_clean_db(db_engine):
 
     # We need to re-init the DB as it creates the default test group and
     # possibly more in future?
-    db.init(db_engine, authority=TEST_SETTINGS["h.authority"])
-
-
-@pytest.fixture(scope="session")
-def db_engine():
-    db_engine = db.make_engine(TEST_SETTINGS)
-    db.init(db_engine, authority=TEST_SETTINGS["h.authority"], should_create=True)
-
-    yield db_engine
-
-    db_engine.dispose()
+    db.pre_create(db_engine)
+    db.Base.metadata.create_all(db_engine)
+    db.post_create(db_engine)
```

Comment:
This is the functests `with_clean_db()` fixture that's used by certain tests that need a clean DB even though the functests don't generally reset the DB between tests. It's a pain to have to duplicate the hook calling here but :man_shrugging: 

To be honest, I think this thing where h's functests don't reset the DB between tests is a bad idea. It makes h's functests different from h's unit tests and also different from every other app's functests. I think it'd be better to have the functests reset the DB in a fast way like the unit tests do and then run them in parallel. But there's more work to do to get there.